connectors/src/connectors/notion/temporal/workflows/index.ts

```diff
@@ -220,6 +230,34 @@ export async function notionSyncWorkflow({
   });
 }
 
+export async function notionDrainDocumentUpsertQueueWorkflow({
+  connectorId,
+}: {
+  connectorId: ModelId;
+}) {
+  const startTime = Date.now();
+  const timeoutMs = 60 * 60 * 1000; // 1 hour
+
+  // This sleep is to ensure that any activities that were just scheduled have time to start before our first check
+  await sleep(1000);
+
+  while (Date.now() - startTime < timeoutMs) {
+    // Call the activity which polls up to 5 times (5 seconds max)
+    const isDrained = await drainDocumentUpsertQueue({ connectorId });
+
+    if (isDrained) {
+      return;
+    }
+
+    // Queue not yet drained, sleep for 5 seconds before trying again
+    await sleep(5000);
```

Comment:
It's a little arbitrary to have it poll every second for 5 seconds, and then wait 5 seconds without any polling before going again. But this is not an exact science, and I was try to not start activities too often.